### PR TITLE
rewrite of validation mechanism

### PR DESCRIPTION
**Pull request by zoranbosnjak**
There was a conceptual problem with validation function, since it assumes "valid" for all unknown instances. This in turn generates valid status for some otherwise false imputs (false positives). This implementation instead assumes "not implemented" unless "errors" method is implemented on all subitems.

This change introduces small incompatibility with previous version.
- ``is_valid`` function is removed
- ``obj.is_valid property`` is introduced instead
- ``obj.errors() method`` is introduced to get error details

Test cases and README file is changed to reflect the change.

I have only small set of geojson data available. I suggest to run validation tests on bigger dataset if you have it available.

### CODE REVIEW

File: geojson/geometry.py [hunk 1]
```diff
@@ -58,31 +56,52 @@ def __init__(self, geometries=None, **extra):
...
 
 class MultiLineString(Geometry):
-    pass
+    def errors(self):
+        return self.checkListErrors(checkLineString, self['coordinates'])
 
 
-class Polygon(Geometry):
-    pass
+def checkPolygon(coord):
```
Comment (frewsxcv) [inline geojson/geometry.py]: snake case

File: geojson/geometry.py [hunk 2]
```diff
@@ -58,31 +56,52 @@ def __init__(self, geometries=None, **extra):
...
+    lengths = all([len(elem) >= 4 for elem in coord])
```
Comment (frewsxcv) [inline geojson/geometry.py]:
```python
all([len(elem) >= 4 for elem in coord])
```

**Comment by zoranbosnjak** [comment]
I have fixed the comments, except I did not understand last remark:

    -class Polygon(Geometry):
    -    pass
    +def checkPolygon(coord):
    +    lengths = all([len(elem) >= 4 for elem in coord])
    
    all([len(elem) >= 4 for elem in coord])

What needs to be changed here?

**Comment by frewsxcv** [comment]
> What needs to be changed here?

woops, meant to write:

```
all(len(elem) >= 4 for elem in coord])
```

Also, do you happen to know if your changes fix https://github.com/frewsxcv/python-geojson/issues/99 ?

**Comment by zoranbosnjak** [comment]
I have converted from list to generator.

Regarding the issue #99, the validation now raises attribute error.
It turns out that the features attribute is a "dict" in this partucilar case, not GeoJSON instance.

Check this:
```python
import geojson

fc = geojson.loads("""{
    "type": "FeatureCollection",
    "features": [{}]
    }
""")

print type(fc.features[0])
print fc.errors()

>>> <type 'dict'>
```

I am not sure, but it might be better to fix #99 in the __init__, so that features list is a list of Feature instances. The validation would in this case turn False on the Feature instance and also on toplevel FeatureCollection object.

File: geojson/geometry.py [hunk 3]
```diff
@@ -58,31 +56,52 @@ def __init__(self, geometries=None, **extra):
...
 
 
-class Polygon(Geometry):
-    pass
+def check_polygon(coord):
+    lengths = all((len(elem) >= 4 for elem in coord))
+    if lengths is False:
+        return 'LinearRing must contain with 4 or more positions'
+
+    isring = all([elem[0] == elem[-1] for elem in coord])
```
Comment (frewsxcv) [inline geojson/geometry.py]:
```python
isring = all(elem[0] == elem[-1] for elem in coord)
```

**Comment by frewsxcv** [comment]
thanks!

**Comment by frewsxcv** [comment]
just published 2.0.0

**Pull request merged**